### ISSUE TITLE
Challenges Reproducing MonoLSS Performance on the Waymo Dataset

### ISSUE THREAD

**Issue by floriangue** [Traffic-X/MonoLSS#20]
Hello,

I’ve been experimenting with various parameter settings but still struggle to reproduce the performance of MonoLSS on the Waymo dataset, even though I’ve successfully replicated the results on KITTI validation. 

### Training and Testing Parameters
Following your recommendation in #12, I referred to DEVIANT's dataset class for guidance. I also attempted to configure the Waymo parameters similarly to how DEVIANT adjusts its Waymo configuration relative to KITTI. Below is a summary of the parameter configurations:

|parameter | DEVIANT KITTI cfg | DEVIANT Waymo cfg | MonoLSS KITTI cfg | MonoLSS Waymo: my experimental parameters |
| -------- | ------- |  -------- | ------- |  -------- |
| resolution | (1280,384) | (768, 512) | (1280,384) | (768, 512) (also tried 960, 640 but worse results) |
| batch size | 12 | 40 | 16 | 40 |
| random_flip | 0.5 | 0.5 | 0.5 | 0.5 |
| random_crop | 0.5 | 0.5 | 0.5 | 0.5 |
| scale | 0.4 | 0.4 | 0.4 | 0.4 |
| shift | 0.1 | 0.1 | 0.1 | 0.1 |
| MixUp3D Augm | None | None | True | True |
| max epoch | 140 | 30 | 150 | 60 (also tried 30 but worse results) |
| decay list | [90, 120] | [18, 26] | [90, 120] | [36, 48] (also tried [18, 26] but worse results) |
| learning rate | 1.25e-3 | 1.25e-3 | 1e-3 | 1e-3 (4e-4 and 2.5e-3 yield worse results) |
| Warmup Epochs LSS Loss | None | None | 50 | 18 |
| tester thold | 0.2 | 0.1 | 0.2 | 0.2 (also tried 0.1 but worse results) |

Sources for DEVIANT Configurations are
DEVIANT Waymo cfg: https://github.com/abhi1kumar/DEVIANT/blob/main/experiments/run_1050.yaml
DEVIANT KITTI cfg: https://github.com/abhi1kumar/DEVIANT/blob/main/experiments/run_221.yaml

### Results from Experiments
I conducted multiple experiments using the "MonoLSS Waymo: my experimental parameters", but I still could not reproduce the results. These are the results I achieved:
- Tester Threshold 0.2: 
<img width="650" alt="image" src="https://github.com/user-attachments/assets/073ff176-5f1a-4fac-b515-15627c909191">

- Tester Threshold 0.1: Unfortunately, using a threshold of 0.1 yielded even worse results.
<img width="650" alt="image" src="https://github.com/user-attachments/assets/4a517bc4-dc70-4398-a8ae-5999a06ae977">

### Request for Assistance
Could I be missing some key parameters that are critical for achieving better performance on the Waymo dataset? Could you share the specific parameters you used when training MonoLSS on Waymo?

Thanks in advance!

Best regards,
Florian Guenther

**Comment by zhaoylee** [comment]
May I ask if you have solved this problem？
